Cargo.toml

```diff
@@ -12,11 +12,15 @@ the rest of the tokio crates.
 """
 
 [dependencies]
-futures = "0.1.4"
+futures = "0.1"
 log = "0.3"
 mio = "0.6"
 scoped-tls = "0.1.0"
 slab = "0.3"
 
 [dev-dependencies]
 env_logger = { version = "0.3", default-features = false }
+nix = "0.7"
```

Comment:
Hmm, it seems libc lacks `errno`. Bringing that into the test, in addition to all the boilerplate already there (for wrapping `RawFd`) would be over the top, at least for my taste. Also, the test would further lose readability, as the `libc` interfaces seem quite "raw" compared to their `nix` wrappers.